tests/lean/grind/fun_induction_fail.lean

```diff
@@ -0,0 +1,333 @@
+    [prop] cexec s'_2 (com.WHILE b_1 c) s' → cexec s_1 (com.WHILE b_1 c) s'
+    [prop] cexec s_1 c s'_2 → cexec s'_2 (com.WHILE b_1 c) s' → cexec s_1 (com.WHILE b_1 c) s'
+    [prop] cexec s_1 (com.WHILE b_1 c) s'
+    [prop] beval s_1 b_1 = true
+    [prop] beval s b_1 = true → cexec s c1 s' → cexec s' (com.WHILE b_1 c1) s' → cexec s (com.WHILE b_1 c1) s'
+    [prop] cexec s c1 s' → cexec s' (com.WHILE b_1 c1) s' → cexec s (com.WHILE b_1 c1) s'
+    [prop] cexec s' (com.WHILE b c1) s'
+    [prop] cexec s' (com.WHILE b_1 c) s'
+    [prop] cexec s' (com.WHILE b_1 c1) s'
+    [prop] beval s' b = false
+    [prop] beval s' b_1 = false
+    [prop] cexec s_2 (com.WHILE b_2 c_1) s_2
+    [prop] cexec s_3 (com.ASSIGN x a) (update x (aeval s_3 a) s_3)
+  [eqc] False propositions
+    [prop] cexec s (com.WHILE b c1) s'
+    [prop] cexec s (com.WHILE b_1 c) s'
+    [prop] cexec s (com.WHILE b_1 c1) s'
+    [prop] cexec s' (com.WHILE b c1) s' → cexec s (com.WHILE b c1) s'
+    [prop] cexec s' (com.WHILE b_1 c) s' → cexec s (com.WHILE b_1 c) s'
+    [prop] cexec s' (com.WHILE b_1 c1) s' → cexec s (com.WHILE b_1 c1) s'
+    [prop] cexec s c1 s'
+    [prop] cexec s c s'
+    [prop] cexec_bounded fuel' s c1 = some s'
+  [eqc] Equivalence classes
+    [eqc] {c, c1, c_1, com.ASSIGN x a}
+    [eqc] {s'_2, s'', s', update x (aeval s_3 a) s_3, s_2}
+    [eqc] {com.WHILE b c1, com.WHILE b_1 c1, com.WHILE b_2 c_1, com.WHILE b_1 c}
+    [eqc] {b, b_2, b_1}
+    [eqc] {s'_1, s_3, s_1}
+    [eqc] {cexec_bounded fuel' s'_1 (com.WHILE b c1), some s'}
+    [eqc] {cexec_bounded fuel' s c1, some s'_1}
+    [eqc] {beval s_2 b_2, beval s' b, beval s' b_1, false}
+    [eqc] {beval s b, beval s_1 b_1, beval s b_1, beval s'_1 b_1, true}
+  [cases] Case analyses
+    [cases] [6/6]: cexec s'_1 (com.WHILE b c1) s'
+    [cases] [5/6]: cexec s'_2 (com.WHILE b_1 c) s''
+    [cases] [2/6]: cexec s_1 c s'_2
+  [ematch] E-matching patterns
+    [thm] beval.eq_2: [beval #0 `[bexp.FALSE]]
+    [thm] beval.eq_1: [beval #0 `[bexp.TRUE]]
+    [thm] cexec_bounded.eq_2: [cexec_bounded (#0 + 1) #1 `[com.SKIP]]
+    [thm] cexec_bounded.eq_1: [cexec_bounded `[0] #1 #0]
+    [thm] cexec.cexec_skip: [cexec #0 `[com.SKIP] #0]
+    [thm] cexec.cexec_while_done: [cexec #3 (com.WHILE #2 #1) #3]
+    [thm] cexec.cexec_while_loop: [cexec #7 (com.WHILE #6 #5) #3, cexec #7 #5 #4]
+    [thm] cexec_bounded.eq_6: [cexec_bounded (#2 + 1) #3 (com.WHILE #1 #0)]
+    [thm] cexec_bounded.eq_3: [cexec_bounded (#2 + 1) #3 (com.ASSIGN #1 #0)]
+    [thm] update.eq_1: [update #3 #2 #1 #0]
+    [thm] cexec.cexec_assign: [com.ASSIGN #1 #0, aeval #2 #0]
+[grind] Diagnostics
+  [thm] E-Matching instances
+    [thm] cexec.cexec_while_loop ↦ 5
+    [thm] cexec.cexec_while_done ↦ 2
+    [thm] cexec.cexec_assign ↦ 1
+-/
+#guard_msgs in
+theorem issue4 : ∀ fuel s c s',
+    cexec_bounded fuel s c = .some s'
+  → cexec s c s' := by
+  intros
+  fun_induction cexec_bounded
+  all_goals grind
```

Comment:
@wkrozowski, could you try to work out how to get to a contradiction from the state `grind` reached here? I think this example is too big to do much with otherwise.